iceoryx2-bb/container/src/byte_string.rs

```diff
@@ -263,15 +262,17 @@ impl<const CAPACITY: usize, const BYTE_CAPACITY: usize> From<&[u8; BYTE_CAPACITY
     }
 }
 
-impl<const CAPACITY: usize> From<&str> for FixedSizeByteString<CAPACITY> {
-    fn from(value: &str) -> Self {
+impl<const CAPACITY: usize> TryFrom<&str> for FixedSizeByteString<CAPACITY> {
+    type Error = FixedSizeByteStringModificationError;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
         if CAPACITY < value.len() {
-            fatal_panic!(from "FixedSizeByteString::from<&str>()", "The string does not fit into the FixedSizeByteString");
+            return Err(FixedSizeByteStringModificationError::InsertWouldExceedCapacity);
```

Comment:
Please never use `return Err(...)` directly, always use the `fail!` macro instead. This way we get for every `Err` - even when it is handled - at least a debug output.
In this case you would write:

```rust
fail!(from "FixedSizeByteString::from<&str>()",
  with FixedSizeByteStringModificationError::InsertWouldExceedCapacity,
  "The provided string \"{}\" does not fit into the FixedSizeByteString with capacity {}",
  str, capacity);
```